### PR TITLE
Added examples test app activity descriptions

### PR DESCRIPTION
**Pull request by langsmith**
## Description

Resolves https://github.com/mapbox/mapbox-navigation-android/issues/2806 by adding descriptions to the tops of the `examples` test app activities.

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### CODE REVIEW

File: examples/src/main/java/com/mapbox/navigation/examples/core/DebugMapboxNavigationKt.kt [hunk 1]
```diff
@@ -65,6 +65,10 @@ import kotlinx.coroutines.channels.Channel
 import okhttp3.Cache
 import timber.log.Timber
 
+/**
+ * This activity shows debug points for raw(red) and
+ * enhanced(blue) location updates.
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/DebugMapboxNavigationKt.kt:70]: This is currently broken 😬 I believe the points are hidden by other layers but currently they're not visible. Totally unrelated to this PR cc @Lebedinsky

File: examples/src/main/java/com/mapbox/navigation/examples/core/FeedbackButtonActivity.kt [hunk 2]
```diff
@@ -43,6 +43,12 @@ import com.mapbox.navigation.ui.utils.ViewUtils
 import java.lang.ref.WeakReference
 import kotlinx.android.synthetic.main.activity_feedback_button.*
 
+/**
+ * This example shows how to integrate the [com.mapbox.navigation.ui.FeedbackButton]'s
+ * feedback report flow with the the Navigation SDK. Road closures and
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/FeedbackButtonActivity.kt:48]:
```suggestion
 * feedback report flow with the Navigation SDK. Road closures and
```

File: examples/src/main/java/com/mapbox/navigation/examples/core/FeedbackButtonActivity.kt [hunk 3]
```diff
@@ -43,6 +43,12 @@ import com.mapbox.navigation.ui.utils.ViewUtils
...
+ * traffic incidents are just two of the several types of live navigation
+ * feedback that can be reported to Mapbox.
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/FeedbackButtonActivity.kt:50]:
```suggestion
 * feedback that can be reported.
```

File: examples/src/main/java/com/mapbox/navigation/examples/core/FreeDriveNavigationActivity.kt [hunk 4]
```diff
@@ -31,7 +31,14 @@ import kotlinx.android.synthetic.main.free_drive_navigation_layout.startNavigati
 import timber.log.Timber
 
 /**
- * To ensure proper functioning of this example make sure your Location is turned on.
+ * This example shows how to enable the Navigation SDK's "free-drive"
+ * mode when the device isn't headed to a specific destination. This
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/FreeDriveNavigationActivity.kt:35]:
```suggestion
 * mode when the device isn't headed to a specific destination (no route is set a.k.a. passive navigation). This
```

File: examples/src/main/java/com/mapbox/navigation/examples/core/ReplayWaypointsActivity.kt [hunk 5]
```diff
@@ -52,7 +52,8 @@ import kotlinx.android.synthetic.main.activity_replay_waypoints_layout.startNavi
 import timber.log.Timber
 
 /**
- * To ensure proper functioning of this example make sure your Location is turned on.
+ * This activity shows how to use navigate to various waypoints
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/ReplayWaypointsActivity.kt:55]: How to use navigate? 🤔

File: examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt [hunk 6]
```diff
@@ -70,6 +70,11 @@ import kotlinx.coroutines.channels.Channel
 import okhttp3.Cache
 import timber.log.Timber
 
+/**
+ * This activity shows how to set up a basic turn-by-turn
+ * navigation experience with many of the the Navigation SDK's
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt:75]: And it's called `SimpleMapboxNavigationKt` 😅 We should definitely rename this example.

File: examples/src/main/java/com/mapbox/navigation/examples/core/TripServiceActivityKt.kt [hunk 7]
```diff
@@ -36,6 +36,12 @@ import kotlinx.coroutines.isActive
 import kotlinx.coroutines.launch
 import timber.log.Timber
 
+/**
+ * This example shows how to use the Navigation SDK's
+ * [MapboxTripService] to push update the Android-system
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/TripServiceActivityKt.kt:41]: To push update the Android-system notification? 🤔

Comment (langsmith) [reply examples/src/main/java/com/mapbox/navigation/examples/core/TripServiceActivityKt.kt]: Fixed

File: examples/src/main/java/com/mapbox/navigation/examples/core/ReplayWaypointsActivity.kt (hunk 5 above)
Comment (langsmith) [reply examples/src/main/java/com/mapbox/navigation/examples/core/ReplayWaypointsActivity.kt:55]: Fixed

**Pull request merged**